beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java

```diff
@@ -286,13 +286,21 @@ public ValidatorApiHandler(
       final UInt64 slot,
       final BLSSignature randaoReveal,
       final Optional<Bytes32> graffiti,
-      final Optional<BeaconState> blockSlotState)
+      final Optional<BeaconState> maybeBlockSlotState)
       throws StateTransitionException {
-    if (blockSlotState.isEmpty()) {
+    if (maybeBlockSlotState.isEmpty()) {
       return Optional.empty();
     }
+    final BeaconState blockSlotState = maybeBlockSlotState.get();
+    final Bytes32 parentRoot = spec.getBlockRootAtSlot(blockSlotState, slot.minus(1));
+    if (!combinedChainDataClient.isFullyValidatedHotBlock(parentRoot)) {
+      LOG.warn(
+          "Unable to produce block at slot {} because parent has optimistically validated payload",
+          slot);
```

Comment:
are we expecting this to be rare? will warning be annoying?

Reply:
Yes that should be rare and I think it's worth knowing if it happens.